tests/integration/ha/test_ha.py

```diff
@@ -107,37 +115,200 @@ async def test_kill_broker_with_topic_leader(
     c_writes_runner: ContinuousWrites,
     restart_delay,
 ):
-    initial_leader_num = await get_topic_leader(
+    topic_description = await get_topic_description(
         ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME
     )
-    initial_offsets = await get_topic_offsets(
-        ops_test=ops_test,
-        topic=ContinuousWrites.TOPIC_NAME,
-        unit_name=f"kafka/{initial_leader_num}",
-    )
+    initial_leader_num = topic_description.leader
+    initial_offsets = await get_topic_offsets(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
 
     logger.info(
         f"Killing broker of leader for topic '{ContinuousWrites.TOPIC_NAME}': {initial_leader_num}"
     )
     await send_control_signal(
-        ops_test=ops_test, unit_name=f"{APP_NAME}/{initial_leader_num}", kill_code="SIGKILL"
+        ops_test=ops_test, unit_name=f"{APP_NAME}/{initial_leader_num}", signal="SIGKILL"
     )
     # Give time for the service to restart
```

Comment:
**question** Should we assert whether the in-sync-replica is not all units anymore?

Reply:
Yes! I can add that here.